File: frontend/src/app/models/Ejercicio.ts
export class Ejercicio {
    public nombre: string;
    public descripcion: string;
    public grupoMuscular: string;
    public equipoNecesario: string;;

    constructor(nombre: string, descripcion: string, grupoMuscular: string, equipoNecesario: string) {
    
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.grupoMuscular = grupoMuscular;
        this.equipoNecesario = equipoNecesario;

    }
}

File: prueba1/src/main/java/org/getfit/entities/Ejercicio.java
package org.getfit.entities;

import java.util.Collection;

import jakarta.persistence.Column;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToMany;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder

@Entity
public class Ejercicio {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(unique = true)
	@NonNull
	private String nombre;
	
	private String descripcion;
	private String grupomuscular;
	private String equiponecesario;

	
	
	@ManyToMany(mappedBy = "ejercicios")
	private Collection<Rutina> rutinas;
}

